### ISSUE TITLE
tls_cert_request generates invalid request for wildcard certs

### ISSUE THREAD

**Issue by saxonww** [terraform-providers/terraform-provider-tls#21]
### Terraform Version

Terraform 0.11.3
TLS provider 1.1.0

### Affected Resource(s)
- tls_cert_request

### Terraform Configuration Files
```hcl
resource "tls_private_key" "ssl_private" {
  algorithm = "RSA"
  rsa_bits  = "2048"
}

resource "tls_cert_request" "ssl_request" {
  key_algorithm   = "${tls_private_key.ssl_private.algorithm}"
  private_key_pem = "${tls_private_key.ssl_private.private_key_pem}"

  subject {
    common_name = "*.example.com"
  }

  dns_names = ["*.example.com", "example.com"]
}

output "ssl_request" {
  value = "${tls_cert_request.ssl_request.cert_request_pem}"
}
```

### Expected Behavior

Terraform generates a valid X.509 certificate request.

### Actual Behavior

The certificate request cannot be used with an OpenSSL CA. An error "The string contains characters that are illegal for the ASN.1 type" is returned. I believe the certificate request is not formatted correctly.

### Steps to Reproduce

1. `terraform apply`
2. `terraform output ssl_request > request.pem`
3. `openssl ca [parameters] -in request.pem -out certificate.pem`

### Important Factoids

I think what's going on here is that Go is encoding the CommonName using the ASN.1 'PrintableString' character string type. The '*' character is not valid for this string type (see [here](https://www.obj-sys.com/asn1tutorial/node128.html)). I verified the string type using [this ](https://lapo.it/asn1js) online ASN.1 decoder; sending another request generated by OpenSSL for the same commonName through the same decoder showed that CommonName was encoded using the 'UTF8String' character string type.

I'm not sure if Terraform can fix this or not. There is an 'ExtraNames' field in [pkix.Name](https://golang.org/pkg/crypto/x509/pkix/#Name) which can override CommonName, but it's not obvious whether you can pass something that Go will encode as a UTF8String instead of a PrintableString.

I was able to override the subject when using the request (e.g. `-subj /CN=\*.example.com`) to work around this. I think it's also valid to have a non-wildcard commonName and have a wildcard entry in the SAN field instead.

**Comment by saxonww** [comment]
I was looking at this very late last night and wasn't thorough. This has already been fixed upstream at golang/go@3b186db7b4a. I have verified that if you build the provider with 1.10.0, `tls_cert_request` generates a valid request even if `*` is in the commonName field.

I'll send a PR with a note added to the documentation, but maybe releasing a new version compiled against 1.10 is something the Terraform team could consider to fix this.

**Comment by mildwonkey** [comment]
Hi! 

We're just released a new version of the tls provider (1.2.0) built with go 1.10 and the documentation has been updated to call out the issue in prior versions of this provider.

Thanks all!

**Comment by lethargosapatheia** [comment]
@mildwonkey Hello. I'm not sure if I've understood this correctly, but are wildcard certificates mentioned in the documentation somewhere? I don't see it here at all: https://registry.terraform.io/providers/hashicorp/tls/latest/docs/resources/cert_request
What is the expected behaviour?
